lib/android-helpers.js

```diff
@@ -373,13 +373,23 @@ helpers.unlock = async function (adb) {
 
   await retryInterval(10, 1000, async () => {
     logger.debug("Screen is locked, trying to unlock");
-    await adb.startApp({
+
+    // first manually stop the unlock activity
+    await adb.forceStop('io.appium.unlock');
+
+    // then start the app twice, as once is flakey
+    let startOpts = {
       pkg: "io.appium.unlock",
       activity: ".Unlock",
       action: "android.intent.action.MAIN",
       category: "android.intent.category.LAUNCHER",
-      flags: "0x10200000"
-    });
+      flags: "0x10200000",
+      stopApp: false
+    };
+    await adb.startApp(startOpts);
+    await adb.startApp(startOpts);
```

Comment:
We cannot fix unlock_apk itself, This issue isn't bug of unlock code, the root cause is the unlock app does not invoke when we call 'am start' after 'force-stop' by ADB

@imurchie,  could you move 'forceStop' out side retryInterval, because retryInterval itself call adb.startApp many time

`await adb.forceStop('io.appium.unlock');

await retryInterval(10, 1000, async () => {
      logger.debug("Screen is locked, trying to unlock");
      let startOpts = {
        pkg: "io.appium.unlock",
        activity: ".Unlock",
        action: "android.intent.action.MAIN",
        category: "android.intent.category.LAUNCHER",
        flags: "0x10200000",
        stopApp: false
      };
      await adb.startApp(startOpts);
})`